src/infarrays.jl

```diff
@@ -202,6 +202,8 @@ end
 # Diagonal
 #####
 
+one(D::Diagonal{T,<:AbstractFill}) where T = Eye{T}(size(D,1))
```

Comment:
This counts as "type piracy": it changes behaviour of FillArrays.jl which is bad.

Instead restrict to the infinite case:
```julia
one(D::Diagonal{T,<:AbstractFill{T,Tuple{OneToInf{Int}}}) where T = Eye{T}(size(D,1))
copy(D::Diagonal{T,<:AbstractFill{T,Tuple{OneToInf{Int}}}) where T = D
```
Though this could be refined to allow for infinite offset indexing.